### DISCUSSION TITLE
Effective usage of Person , Related Persons and Patient in FHIR Core apps

### DISCUSSION THREAD

**Discussion by f-odhiambo** [opensrp/fhircore#764] [Ideas]
We need to define the effective usage of [Person](https://www.hl7.org/fhir/person.html), [RelatedPerson](https://www.hl7.org/fhir/relatedperson.html), [Patient](https://www.hl7.org/fhir/patient.html) resources effectively. We are currently storing all clients as Patients and in the case of a family, modelling a Family using Tags, and extensions to represent family members relationships i.e HHH + Family members 

**Use cases** 

**Adding Clients** 
1. Do all clients become Patients automatically?
2. Do we add a client as a Person and then once a CarePlan is attached to them convert them to a patient? Do we maintain 2 resources? Do we discard the other i.e. Person?
3. When a related person also starts to receive care to they become a Patient?

**Representing a Family** 

The current ANC app stores all family members as Patients either by using tags or extensions to differentiate who is a HHH or a family member.
If we were to effectively do a search for all family members with a given name we could  have something like https://turn-fhir.smartregister.org/fhir/Patient/?family=Bari and get all matching results however this would mean that all family members are added with the same family name, but would not make sense given that you may be  living with a family but not necessarily relates e.g a Guard or Help 

**EIR** 
 In child  Immunizations, we may have a guardian  as a RelatedPerson and the child as a Patient, or in the case of Covax all clients as Patients 

For all the above scenarios we need to figure out a way to make this work seamlessly and also at the point of registration define whether to enrol as a [Person](https://www.hl7.org/fhir/person.html), [RelatedPerson](https://www.hl7.org/fhir/relatedperson.html) or [Patient](https://www.hl7.org/fhir/patient.html) respectively

**Comment by pld** [comment]
I'd say we use Patients everywhere, If we're using groups (or something equally expressive) to record when a Patient is assigned to a HealthCare workflow, we should be able to filter appropriately and we don't have to bother with the conversions you mentioned.

I do think this would put us in a grey area of by the book FHIR, because some of the people we are storing as Patients wouldn't be receiving or expected to receive care. However, it's also justifiable to use a Patient for Personal Health tracking, and that seems like a sufficient justification for using a Patient instead of the other people resources.

tl;dr let's use the Patient resource everywhere
